src/models/Zone.java

```diff
@@ -182,15 +182,15 @@ private static boolean containsClock(Clock clock, List<Clock> clocks) {
         return clocks.contains(clock);
     }
 
-    public Guard buildGuardsFromZone(List<Clock> clocks, List<Clock> relevantClocks) {
-        List<Guard> guards = new ArrayList<>();
-        guards.addAll(buildNormalGuardsFromZone(clocks, relevantClocks));
-        guards.addAll(buildDiagonalConstraintsFromZone(clocks, relevantClocks));
-        return new AndGuard(guards);
+    public Expression buildExpressionFromZone(List<Clock> clocks, List<Clock> relevantClocks) {
```

Comment:
Using "build" implies the builder pattern, and we can remove "FromZone" as it is a member of the ``Zone`` class. We should consider another name like ``createExpression``.